tools/codegen/api/types.py

```diff
@@ -43,6 +100,9 @@ def cpp_type(self, *, strip_ref: bool = False) -> str:
             return self.elem.cpp_type(strip_ref=strip_ref)
         return f'const {self.elem.cpp_type()} &'
 
+    def cpp_type_registration_declarations(self) -> str:
+        return f'const {self.elem.cpp_type_registration_declarations()} &'
```

Comment:
I don't know how much appetite you have for tweaking this code, but if you added `strip_namespace` to `cpp_type()` alongside `strip_ref`, you could do away with all these extra methods...